net/http/authz/store.go

```diff
@@ -40,38 +40,45 @@ func (s *Store) Load(ctx context.Context, policy []string) ([]*Grant, error) {
 	return grants, nil
 }
 
-// Save returns an Op to store g.
-// If a grant equivalent to g is already stored,
-// the returned Op has no effect.
-// It also sets field CreatedAt to the time g is stored (the current time),
-// or to the time the original grant was stored, if there is one.
-func (s *Store) Save(ctx context.Context, g *Grant) sinkdb.Op {
-	key := s.keyPrefix + g.Policy
-	if g.CreatedAt == "" {
-		g.CreatedAt = time.Now().UTC().Format(time.RFC3339)
+// Save returns an Op to store all the grants passed in as g,
+// which must all have the same policy
+// Duplicates are ignored
+// It also sets field CreatedAt to the time the grants
+// are stored, or the time the original grant was stored
+func (s *Store) Save(ctx context.Context, g ...*Grant) sinkdb.Op {
+	if len(g) == 0 {
+		return sinkdb.Op{}
 	}
-
-	var grantList GrantList
-	ver, err := s.sdb.Get(ctx, key, &grantList)
+	policy := g[0].Policy
+	key := s.keyPrefix + policy
+	var existing GrantList
+	ver, err := s.sdb.Get(ctx, key, &existing)
 	if err != nil {
 		return sinkdb.Error(errors.Wrap(err))
 	}
-
-	grants := grantList.Grants
-	for _, existing := range grants {
-		if EqualGrants(*existing, *g) {
-			// this grant already exists, do nothing
-			g.CreatedAt = existing.CreatedAt
-			return sinkdb.Op{}
+	var newGrants []*Grant
+	for _, grant := range g {
+		if grant.Policy != policy {
+			return sinkdb.Error(errors.New("Grants have mismatching policies"))
```

Comment:
No preference, just, if you need to format the string then `fmt.Errorf(...)` is more convenient than `errors.New(fmt.Sprintf(...))`.